### ISSUE TITLE
Emacs style code formatting

### ISSUE THREAD

**Issue by horkhe** [ignatov/intellij-erlang#10]
Emacs code formatter is both _de facto_ and _de jure_ the industry standard. I believe that people will be more eager to move to your IDE if it formats the code the way they used to.

**Comment by artburkart** [comment]
Does this still use emacs? I'm trying to take a stab at a formatter, but it's not as simple as I'd hoped.

**Comment by deadok22** [comment]
Yes, we have an action for formatting with emacs, as well as our own formatter.

**Comment by amir343** [comment]
Hi,

It might be a stupid mistake by me but when I try "Reformat code with Emacs", the whole code is indented with 4 spaces. I even tried to set indent level. Any idea?

**Comment by ignatov** [comment]
Please see (erlang-indent-current-buffer) in (erlang-mode).

**Comment by amir343** [comment]
I mean in IntelliJ IDEA, when I use Code -> "Reformat Code with Emacs", the indentation is with 4 spaces. Do you mean that I need to fix it in Emacs itself?

**Comment by NobbZ** [comment]
Yes, because it runs in an hidden emacs. It's called indend WITH emacs, not
LIKE emacs.

αmir μoulavi <notifications@github.com> schrieb am Mo., 14. März 2016 21:45:

**Comment by amir343** [comment]
Ok, I set erlang-indent-level in my .emacs file and I tested to see if the erlang-indent-current-buffer works and it did. However the IntelliJ still format the code differently. What am I doing wrong here?

**Comment by amir343** [comment]
I was reading the code in "src/org/intellij/erlang/actions/ErlangEmacsFormatAction.java" and I was wondering the way java handles processes, does it read .emacs file from home directory?
